UnitTests/UnitTests.UWP/UI/Collection/Test_IncrementalLoadingCollection.cs

```diff
@@ -34,10 +37,42 @@ public void ConcurentRequests()
         {
             const int pageSize = 20;
             const int pages = 5;
-            var collection = new IncrementalLoadingCollection<DataSource<int>, int>(new DataSource<int>(Enumerable.Range(0, pageSize * pages)), pageSize);
+
+            var source = new DataSource<int>(Enumerable.Range(0, pageSize * pages), DelayedFirstPageSequenc);
+            var collection = new IncrementalLoadingCollection<DataSource<int>, int>(source, pageSize);
+
+
+            var requests = new List<Task>();
+
+            for (int pageNum = 1; pageNum <= pages; pageNum++)
+            {
+                requests.Add(collection.LoadMoreItemsAsync(0).AsTask().ContinueWith(t =>
+                {
+                    Assert.AreEqual(TaskStatus.RanToCompletion, t.Status);
+                    Assert.AreEqual((uint)pageSize, t.Result.Count);
+                }));
+            }
+
+            Task.WaitAll(requests.ToArray());
+            CollectionAssert.AreEquivalent(Enumerable.Range(0, pageSize * pages).ToArray(), collection);
+        }
+
+        [TestMethod]
+        public void FirstRequestFails()
+        {
+            const int pageSize = 20;
+            const int pages = 5;
+
+            var source = new DataSource<int>(Enumerable.Range(0, pageSize * pages), page => throw new Exception());
+            var collection = new IncrementalLoadingCollection<DataSource<int>, int>(source, pageSize);
 
             var requests = new List<Task>();
 
+            requests.Add(collection.LoadMoreItemsAsync(0).AsTask().ContinueWith(t =>
+            {
+                Assert.AreEqual(TaskStatus.Faulted, t.Status);
+            }));
+
             for (int pageNum = 1; pageNum <= pages; pageNum++)
```

Comment:
We should probably wait before we add more data and then check there's no data in the destination?